### PR TITLE
fix: logger on Windows

### PR DESCRIPTION
**Pull request by bzp2010**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

The following errors will appear when the current Manager API is running on windows.

> `panic: couldn't open sink "C:\\apisix-dashboard\\api\\logs\\error.log": no sink found for scheme "c"`

`uber-go/zap` can't directly handle the windows path starting with, for example, `C:\`. This PR adds a new scheme processor, which supports accessing log files on Windows system by using `winfile:///C:\`. At the same time, the program will automatically add `winfile` when generating the absolute path on Windows.

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Review by iamayushdas — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]
Please test manually to confirm that there is no problem, since CI does not support Windows currently.

**Comment by bzp2010** [comment]
@nic-chen 
> Please test manually to confirm that there is no problem, since CI does not support Windows currently.

I found that when I remove the code in `conf.go` to get the absolute path of the file, I can directly use the relative path to write the log on windows.

But in order to be compatible with the file path printed in `printInfo`, I added some compromise code, such as `ErrorLogPathAbs`. At the same time, I will refactor the configuration module in another PR (#1946) to solve these problems.

File: api/internal/conf/conf.go [hunk 1]
```diff
@@ -191,22 +192,35 @@ func setConf() {
...
-
-		// access log
-		if config.Conf.Log.AccessLog.FilePath != "" {
-			AccessLogPath = config.Conf.Log.AccessLog.FilePath
+			if runtime.GOOS == "windows" {
+				ErrorLogPath = `winfile:///` + ErrorLogPath
+			}
 		}
 		if !filepath.IsAbs(AccessLogPath) {
+			if strings.HasPrefix(AccessLogPath, "winfile") {
```
Comment (tokers) [inline api/internal/conf/conf.go:214]: AFAIK the go package `path` handles the windows path normally. Why just use functions provided by it? Like `path.Abs`.

Comment (bzp2010) [reply api/internal/conf/conf.go:214]: - First of all, the function of getting the absolute path of a file is the original code, and I don't fully understand why I do it.
After my test, I found if I don't get the absolute path here, I can't test it pass on Linux. But using relative path can work normally on Windows.

- The purpose of detecting winfile prefix here is to define absolute path on Windows system in `conf.yaml`. `uber-go/zap` cannot handle paths that start with Windows drive letters such as `C:\`.

**Review by tokers — APPROVED** [review]

**Pull request merged**